src/ui/common/src/utils/data.ts

```diff
@@ -21,6 +21,9 @@ export type DataSchema = {
 
 export type Data = {
   schema?: DataSchema;
+  //data: (string | number | boolean)[][];
+  // data is an array of objects whose keys correspond to the schema above.
+  // each record of data is a row on the table.
   data: any[];
```

Comment:
Ready for some fun?
This is a weird typing issue that I encountered recently only when trying to make some mock data to be used in a table. I followed the type to a T (no pun intended), yet found that nothing was rendered on the table.

Next, I looked at the real payloads that we were rendering in preview, and they were  simply an array of objects, where each object in the array corresponds to a row, and each key in the object corresponds to a column in the table. This is different than the 2d array as described in the type above.

Now, as for why we didn't catch this issue earlier? Well, after getting preview data from the API, we would parse the response with JSON.parse(response), and then cast that as a Data object. 

i.e. ```const parsedData = JSON.parse(response) as Data;```

This doesn't actually get type checked and slipped through as a result 😄 . I need to do some more digging on the Typescript docs as to why, but it was a fun thing to learn at the time.

Last thing I promise :) 

The type for data can be made stronger (using ```any``` is frowned upon), so here's what we can do: use a key value pair where the values are of type string | number | boolean. 

Like so:

```data: {[key: string]: string | number | boolean}```